### ISSUE TITLE
repeated same data

### ISSUE THREAD

**Issue by deepdad** [ghcollin/tftables#1]
It -seems- to send the same data into the graph multiple times. I used a data set from here: http://download.nexusformat.org/sphinx/examples/h5py/#id4 and don't know for sure what is in it, however, putting print statements in tftables' getbatch().readbatch() yield statements, outputs the same data several times. I don't see why this would be expected?

**Comment by ghcollin** [comment]
Right now the library will cycle through the dataset multiple times. I wrote it this way for training, where, once you've got to the end of the dataset, you want to wrap around to the beginning to keep training.

I can see the need for a single pass through the dataset for classification tasks though, so I'll see if I can add an option to have it close the tensorflow queue once it gets to the end of the dataset.

**Comment by deepdad** [comment]
I see, then it would shuffle at the training stage too. If the feed (this is a feed right?) takes care of shuffling, then you may want to add that too, as an option, right? Or does it already shuffle due to multi threading?

**Comment by DolanDack** [comment]
Hi, I am brand new to python and I am trying to use your code to train on tensorflow using datasets other than mnist. When I am following your guides there is this line: `array_batch_placeholder = reader.get_batch('/internal/h5_path/to/array')`

I do not really get what this pathway refers to. Tensorflow is quite different from R and Matlab that I have been used to use until now and I cannot really check the variables by executing small batches of the code. The /internal/h5_path/to/array is it something I should provide? I know that in the: `reader = tftables.open_file(filename='path/to/h5_file', batch_size=20)` I use the h5 file path. 

I tried to understand following the more applicable examples that you are offering but to be honest I feel quite lost.

**Comment by ghcollin** [comment]
The library does not shuffle, but the data might be returned out of order due to the concurrent read processes. 

I'm adding a feature to multitables right now that will force the data to be returned in order, and assuming it doesn't have a massive performance impact, I'll make it the default for tftables (with the option to turn it off for better performance).

Shuffling is somewhat hard to add in a way that get good read performance, but I'll think about if there is a good way to approach it.

**Comment by ghcollin** [comment]
Ability to request on-disk ordering and single pass (non-cyclic) access added in 4b29b3d8ab79d7978a33146de59c0050be642f05
